realm/realm-library/src/androidTestObjectServer/java/io/realm/AuthenticateRequestTests.java

```diff
@@ -73,6 +74,9 @@ public void errorsNotWrapped() {
             fail();
         } catch (ObjectServerError e) {
             assertEquals(ErrorCode.ACCESS_DENIED, e.getErrorCode());
+        } finally {
+            // Reset the auth server implementation for other tests.
+            SyncManager.setAuthServerImpl(originalAuthServer);
```

Comment:
Ahh global state 🤓. This isn't an integration test, because we expect integration tests to run separately in their own process to isolate/minimise impact